aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -315,5 +315,50 @@ Tensor& matmul_out(Tensor &result, const Tensor & tensor1, const Tensor & tensor
   return result;
 }
 
+Tensor matrix_power(const Tensor& a, int64_t n) {
+  AT_CHECK(a.dim() >= 2 && at::isFloatingType(a.type().scalarType()),
+           "matrix_power(", a.type(), "{", a.sizes(), "}): expected a tensor "
+           "of floating types with dim at least 2");
+  if (n == 0) {
```

Comment:
The derivative in this case is supposed to be a matrix of 0s . Is there a way to register the gradient here?

Reply:
I don't see any data dependence on a, so shouldn't this be the case automatically?

Reply:
Unfortunately doesn't seem so, which is why I was puzzled.

Reply:
Specifically, this is the error:
```
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```